code/game/objects/items/indoctrinator.dm

```diff
@@ -0,0 +1,74 @@
+/obj/item/indoctrinator
+	name = "indoctrinator"
+	desc = "What!? Who gave you that? It is NOT POSSIBLE!!!"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "indoctrinator"
+	item_state = "electronic"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	force = 0
+	w_class = WEIGHT_CLASS_SMALL
+	throwforce = 0
+	throw_speed = 3
+	throw_range = 7
+	item_flags = NOBLUDGEON
+	var/given_faction = "Coders" // What faction it gives and checks for.
+	var/item_charges = 999 // How many uses left
+	var/indoc_message = "You are now part of coders. You will code funny PRs." // The message that will show up and added to the notes on use.
+	var/use_time = 4 SECONDS // How long will be the do_after.
+	var/in_use = FALSE
+
+/obj/item/indoctrinator/examine(mob/user)
+	. = ..()
+	. += "It has [item_charges] uses left."
+
+/obj/item/indoctrinator/attack(mob/living/carbon/human/M, mob/living/carbon/human/user)
+	if(user.a_intent == INTENT_HARM) // On harm intent do nothing useful.
+		..()
+		return
+	if(in_use)
+		return
+	if(M == user)
+		to_chat(user, "<span class='warning'>You can't use [src] on yourself!</span>")
+		return
+	if(item_charges <= 0) // No charges left.
+		to_chat(user, "<span class='warning'>[src] has no uses left!</span>")
+		return
+	if(given_faction in M.faction) // Target already has the faction.
+		to_chat(user, "<span class='notice'>[M] is already a part of [given_faction].</span>")
+		return
+	if(M.stat > SOFT_CRIT)
+		to_chat(user, "<span class='warning'>[M] should be healthy to receive treatment!</span>")
+		return
+	if(!M.mind)
+		to_chat(user, "<span class='warning'>[M] has no mind!</span>")
+		return
+	if(!M.key)
+		to_chat(user, "<span class='warning'>[M] isn't capable of receiving treatment.</span>")
+		return
+	if(!M.getorgan(/obj/item/organ/brain))
+		to_chat(user, "<span class='warning'>[M] is lacking a brain to operate on!</span>")
+		return
+	to_chat(user, "<span class='notice'>You begin to place [src] near the head of [M].</span>")
+	. = STOP_ATTACK_PROC_CHAIN
```

Comment:
Re-thinking it, this should go before any `to_chat()` message happens, since those are events already.
Just put this before `if(in_use)`